src/server.js

```diff
@@ -1,48 +1,133 @@
 
-    player.insert(newPlayer)
-    addBodies(createBox(newPlayer.x, newPlayer.y))
-    start = true
+      // Store new player and with its socket ID
+      const newPlayer = {
+        id: socket.id,
+        name,
+        color: utils.randomColor(),
+        x: utils.random(1, 7),
+        y: utils.random(1, 6),
+        latency: 0,
+      }
+      await player.insert(newPlayer)
+      socket.emit(events.SERVER_SET_PLAYER, socket.id)
 
-    // Connect events
-    socket.emit(SET_PLAYERS, await player.getAll())
-    player.onChange(p => io.emit(UPDATE_PLAYER, p))
-    socket.on(UPDATE_PLAYER, player.update) // Should be checked (for cheating player)
+      // Send players to all clients
+      io.emit(events.SERVER_ADD_PLAYERS, await player.getAll())
+    })
+
+    // When the game is started
+    socket.on(events.CLIENT_START_GAME, async () => {
+      // Re-init values if restarted
+      players = []
+      if (physicInterval) clearInterval(physicInterval)
+      if (synchroInterval) clearInterval(synchroInterval)
+
+      // Initialize physic server
+      physic.init()
+      physicInterval = setInterval(physic.tick, (1 / 60) * 1000)
+
+      // Add players to physic engine
+      const playersConnected = await player.getAll()
+      playersConnected.forEach(p => players.push({ ...p, body: physic.createBox(p.x, p.y) }))
+
+      // Set server state to 'STARTED'
+      setGameState(GAME_STARTED)
+
+      // Synchronize clients with server physics
+      synchroInterval = setInterval(async () => {
+        io.emit(events.SERVER_SYNCHRONIZE, players.map(p => ({
+          id: p.id,
+          force: p.body.force,
+          position: p.body.position,
+          velocity: p.body.velocity,
+          angle: p.body.angle,
+          angularForce: p.body.angularForce,
+          angularVelocity: p.body.angularVelocity,
+        })))
+      }, 200)
+    })
+
+    // When client moves a player
+    socket.on(events.CLIENT_MOVE, ({ id, velocity }) => {
+      const { body } = find(players, { id })
+      p2.vec2.add(body.velocity, body.velocity, velocity)
+    })
+
+    // When a client is disconnected
+    socket.on(events.DISCONNECT, async () => {
+      logger.info(`Client disconnected : ${socket.id}`)
+      // delete player from DB
+      await player.delete(socket.id)
+      // check if enought players to play
+      const connectedPlayers = await player.getAll()
+      if (connectedPlayers.length <= 1) setGameState(GAME_WAITING)
+      // Send players to all clients
+      io.emit(events.SERVER_ADD_PLAYERS, connectedPlayers)
+    })
+
+    // When a client responds to a ping
+    socket.on(events.CLIENT_PONG, async (startTime) => {
+      const latency = Math.floor((new Date().getTime() - startTime) / 2)
+      const p = await player.get(socket.id)
+      await player.update({ ...p, latency })
+      io.emit(events.SERVER_ADD_PLAYERS, await player.getAll()) // FIXME use DB onChange
```

Comment:
You will overload the network : each user will send all client to each client.
With 100 players, this is 100*100 requests, only for a ping, each second.

Maybe just user an other `setInterval()` to send all "pong" values is enough.